folium/features.py

```diff
@@ -328,15 +343,58 @@ def style_function(x):
                 return {}
         self.style_function = style_function
 
+        self._popup_function = None
+        self.popup_attribute = None
+        self.popup_function = popup_function  # Set the popup function
+
         self._template = Template(u"""
             {% macro script(this, kwargs) %}
+                {% if this.popup_attribute or this.popup_function %}
+                function {{this.get_name()}}_EachFeature(feature, layer) {
+                    if (feature.properties && feature.properties.{{this.popup_attribute}}) {
+                        layer.bindPopup(feature.properties.{{this.popup_attribute}});
+                    }
+                }
+                {% endif %}
                 var {{this.get_name()}} = L.geoJson(
-                    {% if this.embed %}{{this.style_data()}}{% else %}"{{this.data}}"{% endif %})
+                    {% if this.embed %}{{this.style_data()}}{% else %}"{{this.data}}"{% endif %},
+                    {{'{'}}
+                    {% if this.popup_attribute or this.popup_function %}
+                        onEachFeature: {{this.get_name()}}_EachFeature
+                    {% endif %}
+                    {{'}'}}
+                    )
                     .addTo({{this._parent.get_name()}});
                 {{this.get_name()}}.setStyle(function(feature) {return feature.properties.style;});
             {% endmacro %}
             """)  # noqa
 
+    @property
+    def popup_function(self):
+        return self._popup_function
+
+    @popup_function.setter
+    def popup_function(self, func):
+        """
+        Use a setter so that if the dev sets the property function after the
+        fact the `popup_attribute` is set correctly.
+
+        Parameters
+        ----------
+        func : srting, function or None
+            The popup value for the feature
+            * If string, then an attribute to use as a popup value for the feature
+            * If function, a function taking a GeoJson Feature and returning a
+              html string to be used as the popup
+        """
+        if isinstance(func, FunctionType):
+            self.popup_attribute = "_popupContent"
+        elif isinstance(func, text_type):
+            self.popup_attribute = func
+        elif isinstance(func, binary_type):
+            self.popup_attribute = text_type(func, 'utf8')
+        self._popup_function = func
+
```

Comment:
This is great, but it seems complicated to a dummy programmer like me.
Can we simplify this in having a single attibute `popup` (playing the role of both `popup_function` and `popup_attribute` and in testing in the rendering whether it's a function or a string ?
It's a question (I try to keep the code as simple as possible) but your PR is ok anyway.